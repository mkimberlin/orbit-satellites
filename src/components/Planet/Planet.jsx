import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './styles.module.css';

const displayName = 'Planet';
const propTypes = {
  label: PropTypes.string,
  color: PropTypes.shape({
    r: PropTypes.number,
    g: PropTypes.number,
    b: PropTypes.number,
    a: PropTypes.number
  }),
  size: PropTypes.oneOf([1, 2, 3]),
  onClick: PropTypes.func
};

const Planet = ({ label, color, size = 2, onClick }) => {
  const styles = {
    backgroundColor: color
      ? `rgba(${color.r}, ${color.g}, ${color.b}, ${color.a})`
      : undefined,
    boxShadow: color
      ? `0 0 1rem rgba(${color.r}, ${color.g}, ${color.b}, ${color.a})`
      : undefined
  };

  return (
    <div
      id="planet"
      className={cx('planet', `planet-${size}`)}
      onClick={onClick}
      style={styles}
    />
  );
};

Planet.displayName = displayName;
Planet.propTypes = propTypes;

export default Planet;
