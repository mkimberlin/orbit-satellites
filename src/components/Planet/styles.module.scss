@import '../../styles/common-variables.scss';

.planet {
  text-align: center;
  color: white;
  background-color: rgba($default-planet-color, 0.7);
  position: absolute;
  border-radius: 50%;
  box-shadow: 0 0 1rem rgba($default-planet-color, 0.7);
  overflow: hidden;
  text-overflow: ellipsis;

  -webkit-transform: rotateY($tilt-y * -1) rotateX($tilt-x * -1);
  transform: rotateY($tilt-y * -1) rotateX($tilt-x * -1);

  @for $planet-size from 1 through $object-sizes {
    &.planet-#{$planet-size} {
      $calculated-size: $planet-size * $planet-step;
      width: $calculated-size;
      height: $calculated-size;
      line-height: $calculated-size;
      top: calc(50vh - (#{$calculated-size / 2}));
      left: calc(50vw - (#{$calculated-size / 2}));
    }
  }
}
