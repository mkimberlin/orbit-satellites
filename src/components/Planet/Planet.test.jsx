import React from 'react';
import Planet from './Planet';

describe('Planet component', () => {
  it('renders correctly with default props', () => {
    snapshot(<Planet />);
  });

  it('renders correctly with alternate props', () => {
    snapshot(
      <Planet
        label="Saturn"
        color={{ r: 255, g: 205, b: 155, a: 0.5 }}
        size={3}
        onClick={jest.fn()}
      />
    );
  });
});
