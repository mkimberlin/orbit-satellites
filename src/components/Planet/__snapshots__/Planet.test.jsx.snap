// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planet component renders correctly with alternate props 1`] = `
<div
  className="planet planet-3"
  id="planet"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 205, 155, 0.5)",
      "boxShadow": "0 0 1rem rgba(255, 205, 155, 0.5)",
    }
  }
/>
`;

exports[`Planet component renders correctly with default props 1`] = `
<div
  className="planet planet-2"
  id="planet"
  onClick={undefined}
  style={
    Object {
      "backgroundColor": undefined,
      "boxShadow": undefined,
    }
  }
/>
`;
